backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/CodeGenerator.kt

```diff
@@ -292,8 +292,11 @@ internal class FunctionGenerationContext(val function: LLVMValueRef,
         LLVMBuildStore(builder, value, ptr)
     }
 
-    fun storeAny(value: LLVMValueRef, ptr: LLVMValueRef) {
+    fun storeAny(value: LLVMValueRef, ptr: LLVMValueRef, shallFreeze: Boolean = false) {
```

Comment:
`shallFreeze` seems to be not that closely related to storing a value to memory. It makes some sense to provide `freeze` function here and call it where required.